stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -313,6 +316,8 @@ pub struct NakamotoBlockHeader {
     pub miner_signature: MessageSignature,
     /// Schnorr signature over the block header from the signer set active during the tenure.
     pub signer_signature: ThresholdSignature,
+    /// A bitvec which represents the signers that participated in this block signature.
+    pub signer_bitvec: BitVec,
```

Comment:
I'm a little skeptical that we should be using the generic `BitVec` here.  Yes, we need a bitmap of some kind, but I think we also need a hard bound on how big this is allowed to be (e.g. there can be no more than 4000 bits).  And, I think that bound needs to be expressed in the data type itself, so it's impossible to represent a block header that would exceed this bound (and thus it's impossible to both serialize and deserialize such a block).

Reply:
I added a generic const to the `BitVec` struct for `MAX_SIZE`. The field in the block header sets this to 4000.